src/main/flags.sh

```diff
@@ -49,7 +49,16 @@ OS="$1"
 # - we're on alpine, in which case the opam switch doesn't have a special
 #   name. It is assumed that the reason we're on alpine is to get
 #   statically-linked executables.
-if [[ "$(opam switch show)" == *+static* ]]; then
+
+
+# Check if NIX_ENVIRONMENT is not set
+# If it is set, we are in a nix-shell and we should not statically link
+# This is first because opam won't exist in a nix build environment
+if [[ -n "${SEMGREP_NIX_BUILD-}" ]]; then
```

Comment:
```suggestion
if [[ ! -v SEMGREP_NIX_BUILD ]]; then
```
can we just use `-v` here for consistency given we require bash 5+ anyway?

Reply:
(untested)

Reply:
trying

Reply:
yea GHA uses bash 4.2 or smthn, workflows failed, force pushing back down to prev commit